include/fem-dem/gls_sharp_navier_stokes.h

```diff
@@ -303,14 +306,40 @@ class GLSSharpNavierStokesSolver : public GLSNavierStokesSolver<dim>
   void
   refine_ib();
 
+
+  /**
+   * @brief
+   *This function checks if all particles are spheres. If one of them is not,
+   *the code will use the regular generate_cut_cells function. If all particles
+   *are spheres, it uses the optimized_generate_cut_cells.
+   */
+  void
+  check_particles_all_sphere();
+  /**
+   * @brief
+   *This function create a map (cut_cells_map) that indicates if a cell is
+   *cut, and the particle id of the particle that cut it.
+   */
+  void
+  generate_cut_cells_map();
+
   /**
    * @brief
    *This function create a map (cut_cells_map) that indicates if a cell is cut,
    *and the particle id of the particle that cut it.
+   *The algorithm was optimized to reduce the cost of this step.
```

Comment:
I would specify that it only applied to spheres which is why we need also the old function.